synapse/http/proxyagent.py

```diff
@@ -208,30 +210,46 @@ def request(self, method, uri, headers=None, bodyProducer=None):
         )
 
 
-def _http_proxy_endpoint(proxy, reactor, **kwargs):
+def _http_proxy_endpoint(
+    proxy: Optional[bytes],
+    reactor: IReactorCore,
+    tls_options_factory: Optional[IPolicyForHTTPS],
+    **kwargs
+):
     """Parses an http proxy setting and returns an endpoint for the proxy
```

Comment:
Can we add a bit about the forms of `proxy` which are supported?

Reply:
Done